pipeline/ingest/util.py

```diff
@@ -0,0 +1,528 @@
+            if isinstance(d['variables'][varname], (list, tuple)):
+                templist = list()
+                for idx in range(0, len(df)):
+                    templist.append(d['variables'][varname])
+                df['var:' + varname] = templist
+            else:
+                df['var:' + varname] = d['variables'][varname]
+    # updating variables
+    variableidxs = (df[df['MSG'] == 'Variables updated:']).index.to_numpy()
+    for variableidx in variableidxs:
+        d = {}
+        exec('variables = ' + df['MSG'][variableidx + 1], d)
+        for varname in d['variables'].keys():
+            if isinstance(d['variables'][varname], (list, tuple)):
+                templist = list()
+                idxs = list()
+                for idx in range(variableidx, len(df)):
+                    idxs.append(idx)
+                    templist.append(d['variables'][varname])
+                df['var:' + varname][variableidx:] = templist.copy()
+            # =============================================================================
+            #                 print(len(templist))
+            #                 print(len(idxs))
+            #                 print(templist)
+            #                 df.loc[idxs, 'var:'+varname] = templist
+            # =============================================================================
+            else:
+                # df['var:'+varname][variableidx:] = d['variables'][varname]
+                df.loc[range(variableidx, len(df)), 'var:' + varname] = d['variables'][varname]
+
+    # saving motor variables (if any)
+    variableidx = (df[df['MSG'] == 'LickportMotors:']).index.to_numpy()
+    if len(variableidx) > 0:
+        d = {}
+        exec('variables = ' + df['MSG'][variableidx + 1].values[0], d)
+        for varname in d['variables'].keys():
+            df['var_motor:' + varname] = d['variables'][varname]
+    # extracting reward probabilities from variables
+    if ('var:reward_probabilities_L' in df.columns) and ('Block_number' in df.columns):
+        probs_l = df['var:reward_probabilities_L'][0]
+        probs_r = df['var:reward_probabilities_R'][0]
+        df['reward_p_L'] = np.nan
+        df['reward_p_R'] = np.nan
+        if ('var:reward_probabilities_M' in df.columns) and ('Block_number' in df.columns):
+            probs_m = df['var:reward_probabilities_M'][0]
+            df['reward_p_M'] = np.nan
+        for blocknum in df['Block_number'].unique():
+            if not np.isnan(blocknum):
+                df.loc[df['Block_number'] == blocknum, 'reward_p_L'] = probs_l[int(blocknum - 1)]
+                df.loc[df['Block_number'] == blocknum, 'reward_p_R'] = probs_r[int(blocknum - 1)]
+                if ('var:reward_probabilities_M' in df.columns) and ('Block_number' in df.columns):
+                    df.loc[df['Block_number'] == blocknum, 'reward_p_M'] = probs_m[int(blocknum - 1)]
+    return df
+
+
+def loadcsvdata(bigtable=pd.DataFrame(),
```

Comment:
Everything from line 213 is not relevant to the datajoint pipeline, it's useful only for online analysis.
you can get rid of it